src/utils.c

```diff
@@ -115,7 +115,7 @@ void * avifArrayPushPtr(void * arrayStruct)
 {
     uint32_t index = avifArrayPushIndex(arrayStruct);
     avifArrayInternal * arr = (avifArrayInternal *)arrayStruct;
-    return &arr->ptr[index * arr->elementSize];
+    return &arr->ptr[index * (size_t)arr->elementSize];
```

Comment:
optional: make the index computation into a function or macro so that the cast doesn't have to be repeated.